### PR TITLE
Upgrade Selenium to 2.53.0 #5829

### LINKED ISSUE #5829: Upgrade Selenium to 2.53.0
The last update was more than a year ago. This version of Selenium has many fixes (7 minor versions worth) and also supports Firefox up to version 46.0 (it still supports 38.0.5 as well so current devs don't need to upgrade if they don't want to).
There is version 2.53.1 but no Java update is made on that patch version.

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #5829 

Unlike the previous upgrade in #3645, much more needs to be done here, particularly because:
- `element.click()` doesn't work in many cases due to the "Element is not clickable at point (X, Y)" error. This necessitates migrating all `element.click()` to `click(element)` which is a self-defined function using javascript event to click the element, and makes up the bulk of change in this PR. Good news is, this is the most major cause for why we can't use Chrome for testing before and it is fixed.
- `assertEquals(status, page.getStatus())` -> `page.verifyStatus(status)`, likely because optimisations have been done on Selenium's part and they try to verify much faster than before.
- The new version of the library doesn't bundle NekoHTML anymore, so we move to Jsoup which is also used in some email service classes. There is no noticeable difference in output.

### CODE REVIEW

**Comment by wkurniawan07** [comment]
Tried and worked on Mac OS X 10.11.2, Firefox 46.0. Yet to try on my Windows environment.

@kanghj @JosephineKwa @YongJieYongJie @chowyb will be good if you can try as well! Just go to this branch, update the libraries with the usual `resetEclipseDeps` command, and run a full test once. Also appreciate if you can use different versions (any one from 38.0.5 to 46.0) of Firefox!

**Comment by chowyb** [comment]
@wkurniawan07 my non-testing Firefox is on 46.0.1 D: I wonder if it'll still work haha

**Comment by kanghj** [comment]
Works for me

**Comment by chowyb** [comment]
Windows 8, Firefox 46.0.1, works for me, there's a constantly failing test case for me but it isn't a UI one. Fails on master on Firefox 38.0.5 (old settings) as well so this isn't caused by this PR.

failed

**Comment by wkurniawan07** [comment]
While waiting for more (good) news, this can call for a proper review already.

File: build.gradle [hunk 1]
```diff
@@ -77,7 +77,8 @@ dependencies {
                     "com.google.appengine:appengine-testing:${appengineVersion}",
                     "httpunit:httpunit:1.7",
                     "javax.mail:mail:1.4.5",
-                    "org.seleniumhq.selenium:selenium-java:2.46.0",
+                    "org.apache.commons:commons-lang3:3.4",
```
Comment (kanghj) [inline build.gradle]: What's this used for? I can't find where it's used

Comment (wkurniawan07) [inline build.gradle]: Look for `import org.apache.commons.lang3.ArrayUtils` and `import org.apache.commons.lang3.StringUtils`.

**Comment by JosephineKwa** [comment]
Sorry this slipped my mind! It works for me when using firefox 38.0.5

**Comment by wkurniawan07** [comment]
@chowyb I got the failure you mentioned, then I ran Godmode, noticed some changes in Git status, but when staging the changes they're all gone just like that (nothing to commit) and subsequent tests pass without Godmode. You can do the same.
(and I don't know exactly what happened, but it should have something to do with the line endings)

**Pull request merged**